imageio/imageio-batik/src/main/java/com/twelvemonkeys/imageio/plugins/svg/SVGImageReader.java

```diff
@@ -115,6 +118,24 @@ public void setInput(Object pInput, boolean seekForwardOnly, boolean ignoreMetad
             TranscoderInput input = new TranscoderInput(IIOUtil.createStreamAdapter(imageInput));
             rasterizer.setInput(input);
         }
+        else if (pInput instanceof Document) {
+            Document doc = (Document) pInput;
+            Element root = doc.getDocumentElement();
+            if (root == null) {
+                processWarningOccurred("Document input has no root element");
+            }
+            else if (root.getLocalName() == null) {
+                processWarningOccurred("Not a namespace-aware Document input: " + root.getTagName());
+            }
+            else if (!root.getLocalName().equals("svg")
+                    || !root.getNamespaceURI().equals(SVG_NS_URI)) {
+                processWarningOccurred("Not a SVG Document input: " + root.getTagName()
+                        + " / {" + root.getNamespaceURI() + "}" + root.getLocalName());
+            }
```

Comment:
Not sure if this sanity check should go here or in the `SVGImageReaderSpi.canDecodeInput()`.  My rationale is the Batik library might support more than `{http://www.w3.org/2000/svg}svg` documents, hence this is just a warning.  In any case, it would signal unsupported types during the image reading/rasterizing.